### PR TITLE
Issue 501 non linear node unscented

### PR DESCRIPTION
**Pull request by Raphael-Tresor**
SOLVE ISSUE: https://github.com/ReactiveBayes/ReactiveMP.jl/issues/501

Unscented() approximation does not work when the input function is an univariate function that returns a vector.

The issue was that `Unscented()`  was dispatching to an `unscented-statistics` method that was not managing such a case.
We dispatch in `unscented-statistics` according to the output.

### CODE REVIEW

**Comment by wouterwln** [comment]
Just to practice TDD: Can you guys add some tests testing edge cases? (For example, what happens if I put in a 0 variance gaussian, infinite variance gaussian, NaNs, stuff like this) such that we know that this Unscented implementation is relatively robust? (Or at least the behaviour is documented and known)

Good job guys!!

**Comment by bvdmitri** [comment]
@wouterwln @Nimrais ping

I personally agree with the proposed behaviour from @Raphael-Tresor

**Comment by wouterwln** [comment]
Yes me too, sounds great!

**Comment by bvdmitri** [comment]
@Raphael-Tresor the CI is failing due to 
```
nested task error: LoadError: Duplicate test item name `Unscented approximation method` in file `test/approximations/unscented_tests.jl` at line 86
```
Each `@testitem` should have a unique name

File: src/approximations/unscented.jl [hunk 1]
```diff
@@ -115,10 +115,22 @@ function statistic_estimation(::Val{C}, first_element::V, g_sigma, sigma_points,
...
 
+__unscented_parameters_zero_covariance(m::T) where {T <: Real} = (m, zero(T), nothing)
+__unscented_parameters_zero_covariance(m::AbstractVector{T}) where {T <: Real} = (m, zeros(T, length(m), length(m)), nothing)
+
 # Single univariate variable
 function unscented_statistics(method::Unscented, ::Val{C}, g::G, means::Tuple{Real}, covs::Tuple{Real}) where {C, G}
     m = first(means)
     V = first(covs)
+    if V == 0.0
+        @warn "Unscented transform called with zero covariance input (function $g)"
```
[the remark this answers is not in the document]
Comment (Raphael-Tresor) [reply src/approximations/unscented.jl:126]: What do you mean?

File: src/approximations/unscented.jl [hunk 2]
```diff
@@ -95,39 +95,75 @@ function unscented_statistics(method::Unscented, g::G, means::Tuple, covs::Tuple
...
+
+__unscented_parameters_zero_covariance(m::T) where {T <: Real} = (m, zero(T), nothing)
+__unscented_parameters_zero_covariance(m::AbstractVector{T}) where {T <: Real} = (m, zeros(T, length(m), length(m)), nothing)
+
 # Single univariate variable
 function unscented_statistics(method::Unscented, ::Val{C}, g::G, means::Tuple{Real}, covs::Tuple{Real}) where {C, G}
     m = first(means)
     V = first(covs)
+    if V == 0.0
+        @warn "Unscented transform called with zero covariance input (function $g)"
```
Comment (wouterwln) [inline src/approximations/unscented.jl:126]:
```suggestion
        @warn "Unscented transform called with zero covariance input (function $g)" maxlog=1
```

**Review by wouterwln — APPROVED** [review]

**Comment by bvdmitri** [comment]
Thanks @Raphael-Tresor !!
